mono/utils/mono-codeman.c

```diff
@@ -57,7 +57,7 @@ static MonoCodeManagerCallbacks code_manager_callbacks;
 #define MAX_WASTAGE 32
 #define MIN_BSIZE 32
 
-#ifdef __x86_64__
+#if defined(__x86_64__) && !defined(__APPLE__)
```

Comment:
FWIW we fixed this a bit differently upstream: https://github.com/dotnet/runtime/commit/cc501c21ee3e91caf2dacac2f06f8cbb0286c710

Reply:
Thanks. Looks like our fork code looks to be quite a bit different. We'll get rid of this change when we pull upstream.